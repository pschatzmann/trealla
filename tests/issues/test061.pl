:- initialization(main).

:- use_module(library(lists)).
:- use_module(library(apply)).

main :-
    maplist(col([[1,2],[3,4]]), [1,2], X1),
    writeln(X1),
    maplist(col([[A,2],[3,A]]), [1,2], X2),
    writeln(X2),
    maplist(length, M, [2,2]),
    writeln(M),
    maplist(col(M), [1,2], X),
    writeln(X).

col(Matrix, N, Column) :-
    maplist(nth1(N), Matrix, Column).
