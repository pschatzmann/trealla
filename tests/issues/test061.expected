[[1,3],[2,4]]
[[_5,3],[2,_5]]
[[_150,_151],[_161,_162]]
[[_150,_161],[_151,_162]]
